### ISSUE TITLE
Quite long running time

### ISSUE THREAD

**Issue by jmlipman** [nstucki/Betti-matching#5]
Hi,

I'm trying to use the loss function on my training pipeline, and I noticed that it takes quite a long time for images of size about 400^2. After 10 minutes of training (batch size=1), it couldn't finish one iteration. I also noticed that in the paper it is reported that experiments were on 48x48 patches. Does it make sense that it takes that long, or am I doing something wrong?

Thanks!

**Comment by nstucki** [comment]
Hi,

an image of size 400^2 is definitely too big to handle for this implementation. That's why we chose a patch size of 48x48 on most datasets.

But if you want to train on bigger patches, I can recommend using our C++ implementation of Betti Matching (https://github.com/nstucki/Betti-Matching-3D).

Best,
Nico
